Source/FieldSolver/ImplicitSolvers/ImplicitSolver_fwd.H

```diff
@@ -0,0 +1,12 @@
+#ifndef _IMPLICIT_SOLVER_FWD_H_
+#define _IMPLICIT_SOLVER_FWD_H_
+
+struct ImplicitOptions {
+    int max_particle_iterations;
+    double particle_tolerance;
+    bool deposit_mass_matrices;
+    bool linear_stage_of_jfnk;
+    int nonlinear_iteration;
```

Comment:
@JustinRayAngus can you please assign default values here, to ensure the struct values are always well-defined?
```suggestion
    int max_particle_iterations = 1;
    double particle_tolerance = WarpX::particle_tol_in_implicit_scheme;
    bool deposit_mass_matrices = false;
    bool linear_stage_of_jfnk = false;
    int nonlinear_iteration = ;
```

Otherwise, if we forget to set a value in usage we will get UB.

Reply:
While you refactor this file you could also use `amrex::ParticleReal` as per the other comment, but works either way for me.